kuma/javascript/src/payments/strings.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+import { gettext, interpolate } from '../l10n.js';
+
+export const strings = {
+    signup: 'Become a monthly supporter',
+    signupDesc:
+        'Support MDN with a $5 monthly subscription and get back more of the knowledge and tools you rely on for when your work has to work.',
+    thankYou: 'Thank you for becoming a monthly supporter!',
+    memberNum: 'You are MDN member number:',
+    potentialMemberNum: 'You will be MDN member number:',
+};
+
+export const getMemberNumberString = (
+    num: ?number,
+    isSubscriber?: ?boolean
+): ?string => {
+    if (isSubscriber && num) {
+        return interpolate(gettext('%(text)s %(num)s'), {
+            text: strings.memberNum,
+            num: num.toLocaleString(),
+        });
+    }
+    if (num) {
+        return interpolate(gettext('%(text)s %(num)s'), {
+            text: strings.potentialMemberNum,
+            num: num.toLocaleString(),
+        });
+    }
+    return null;
```

Comment:
Now, to cake onto that about localization, perhaps it's totally enough to ask translators to translate (only) the string:
```
You will be MDN member number:
```
because of that `:`. It might be worth learning from this and reaching out to @mathjazz or @jotes and ask what they advise. Not having the number there and not having to use `interpolate` at all would make things simpler and less chance of it getting broken. Also, if you're translating to Arabic or Hebrew, does a string like `You are MDN member number:` even make sense to translate without the context of the number because they might want to write: `123 :number member MDN be will You` in their language so they'd need that `%(num)s` as part of the string.

Reply:
@peterbe 
AFAIR, in this case, Gettext will produce the string in this form:
```%(text) %(num)```
If such string lands in a `.po` file, Pontoon is will not pick up the real string that requires a translation (@mathjazz correct me if Gettext is able to handle this in a better way).
Personally, I would try to change this into:
```
gettext('You will be MDN member number: %(num)s')
```
Because Gettext will produce a string which will look like this:
```You will be MDN member number: %(num)s```

To give you some examples of how the translators see such strings:
https://pontoon.mozilla.org/pl/mdn/LC_MESSAGES/django.po/?search=%25s&string=141608

So, I think that you're right about the flexibility of translation and the RTL-related problems. Also
[sidenote] It's worth mentioning that the translators are notified when their translation doesn't include the variables defined in the original string.

Reply:
@jotes is right - Pontoon will show strings as they appear in a `.po` file.

And I agree with his proposal - it's always a good idea to include the `variable` in such string. Both, for more context and less RTL headaches.

Reply:
Thanks everyone-- I'll adopt @jotes' suggestion. 

@jotes @mathjazz, a related question-- do you happen to know why, if a string is passed to a React component as a prop, it must be gettext-ed before? I thought it would simplify things if I had `gettext` being called in one location, which in the example below is the shared header component, instead of in each component that imported it. 

🚫Does not work  
```
// Component that includes `Header`
<Header title="Become a monthly supporter" />

// Header component
<h1>{gettext(title)}</h1>
```

👍 But this does
```
// Component that includes `Header`
<Header title={gettext("Become a monthly supporter")} />

// Header component
<h1>{title}</h1>
```

Reply:
@mindy Hey,

(I'm not an expert in Gettext implementation on React/l10.js stack but maybe my answer will help you)

I think this is caused by the fact that you need a tool (e.g. xgettext) that is able to extract l10n strings from your code.
AFAIR, xgettext scans source files to find all occurrences of `gettext('string')` (sidenote: some people prefer to use the underscore, e.g. ```_('string')```) to extract strings from inside those function calls and puts them in a `.po` file.

Reply:
@mindy I think @jotes is right again. The static analysis tools we have that picks up strings might see `gettext(title)` and see that `title` doesn't look like a valid string because it's not wrapped in quotation marks so it chooses to ignore it when preparing what needs to be translated. 

Either way, I think `<Header title={gettext("Become a monthly supporter")} />` reads much nicer from a code-reader point of view. 

The confusing thing is that how `gettext` works, in runtime, is that it's a simple catalog of strings. 
Go to view-source:https://developer.mozilla.org/fr/search?q=find and find that massive `<script>window._react_data = JSON.parse(...` thing. 
Paste all of that into a Web Console and look at 
```
>> _react_data.stringCatalog
```

So if your English string exists in there (e.g. `!!_react_data.stringCatalog['Become a monthly supporter']`) then doing...
```
var title = "Become a monthly supporter";
return <h1>{gettext(title){</h1>
```
...would work. 
I guess what's explaining things is that we're unable to extract it into the catalog in the first place. 

l10n is hard. Especially when you're not actively working on it.

Reply:
> l10n is hard. Especially when you're not actively working on it.

Trust me: you could easily omit the 2nd sentence. 😆